cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1173,6 +1186,7 @@ deserialiseShelleyBasedTxBody mkTxBody bs =
           (CBOR.runAnnotator txbody fbs)
           (map (`CBOR.runAnnotator` fbs) txscripts)
           (CBOR.runAnnotator <$> txmetadata <*> pure fbs)
+          (error "deserialiseShelleyBasedTxBody: Redeemer pointer map")
```

Comment:
Yep, I wasn't sure how we wanted to handle serialisation.

Reply:
This bit is causing test failures currently.

Reply:
```
    prop_roundtrip_txbody_shelley_CBOR:                        FAIL
        ✗ prop_roundtrip_txbody_shelley_CBOR failed at test/Test/Cardano/Api/Typed/CBOR.hs:156:3
          after 1 test.
        
              ┏━━ test/Test/Cardano/Api/Typed/CBOR.hs ━━━
          152 ┃ roundtrip_CBOR
          153 ┃   :: (SerialiseAsCBOR a, Eq a, Show a)
          154 ┃   => AsType a -> Gen a -> Property
          155 ┃ roundtrip_CBOR typeProxy gen =
          156 ┃   H.property $ do
          157 ┃     val <- H.forAll gen
          158 ┃     H.tripping val serialiseToCBOR (deserialiseFromCBOR typeProxy)
              ┃     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
              ┃     │ ━━━ Exception (ErrorCall) ━━━
              ┃     │ deserialiseShelleyBasedTxBody: Redeemer pointer map
              ┃     │ CallStack (from HasCallStack):
              ┃     │   error, called at src/Cardano/Api/TxBody.hs:1219:12 in cardano-api-1.27.0-inplace:Cardano.Api.TxBody
        
          This failure can be reproduced by running:
          > recheck (Size 0) (Seed 16246837987535537937 6040485580684346801) prop_roundtrip_txbody_shelley_CBOR
        
      Use '--hedgehog-replay "Size 0 Seed 16246837987535537937 6040485580684346801"' to reproduce.
    prop_roundtrip_txbody_allegra_CBOR:                        FAIL
        ✗ prop_roundtrip_txbody_allegra_CBOR failed at test/Test/Cardano/Api/Typed/CBOR.hs:156:3
          after 1 test.
        
              ┏━━ test/Test/Cardano/Api/Typed/CBOR.hs ━━━
          152 ┃ roundtrip_CBOR
          153 ┃   :: (SerialiseAsCBOR a, Eq a, Show a)
          154 ┃   => AsType a -> Gen a -> Property
          155 ┃ roundtrip_CBOR typeProxy gen =
          156 ┃   H.property $ do
          157 ┃     val <- H.forAll gen
          158 ┃     H.tripping val serialiseToCBOR (deserialiseFromCBOR typeProxy)
              ┃     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
              ┃     │ ━━━ Exception (ErrorCall) ━━━
              ┃     │ deserialiseShelleyBasedTxBody: Redeemer pointer map
              ┃     │ CallStack (from HasCallStack):
              ┃     │   error, called at src/Cardano/Api/TxBody.hs:1219:12 in cardano-api-1.27.0-inplace:Cardano.Api.TxBody
        
          This failure can be reproduced by running:
          > recheck (Size 0) (Seed 5526808619565871689 17014309501333539577) prop_roundtrip_txbody_allegra_CBOR
        
      Use '--hedgehog-replay "Size 0 Seed 5526808619565871689 17014309501333539577"' to reproduce.
    prop_roundtrip_txbody_mary_CBOR:                           FAIL
        ✗ prop_roundtrip_txbody_mary_CBOR failed at test/Test/Cardano/Api/Typed/CBOR.hs:156:3
          after 1 test.
        
              ┏━━ test/Test/Cardano/Api/Typed/CBOR.hs ━━━
          152 ┃ roundtrip_CBOR
          153 ┃   :: (SerialiseAsCBOR a, Eq a, Show a)
          154 ┃   => AsType a -> Gen a -> Property
          155 ┃ roundtrip_CBOR typeProxy gen =
          156 ┃   H.property $ do
          157 ┃     val <- H.forAll gen
          158 ┃     H.tripping val serialiseToCBOR (deserialiseFromCBOR typeProxy)
              ┃     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
              ┃     │ ━━━ Exception (ErrorCall) ━━━
              ┃     │ deserialiseShelleyBasedTxBody: Redeemer pointer map
              ┃     │ CallStack (from HasCallStack):
              ┃     │   error, called at src/Cardano/Api/TxBody.hs:1219:12 in cardano-api-1.27.0-inplace:Cardano.Api.TxBody
        
          This failure can be reproduced by running:
          > recheck (Size 0) (Seed 2450485195492175484 8218421726347182459) prop_roundtrip_txbody_mary_CBOR
        
      Use '--hedgehog-replay "Size 0 Seed 2450485195492175484 8218421726347182459"' to reproduce.
```

Reply:
I'll push a fix